File: app/src/main/java/com/example/ruru/sophiesblog/android/kotlin/clazz/Student.java
package com.example.ruru.sophiesblog.android.kotlin.clazz;

import androidx.annotation.NonNull;

public class Student {

  private int id;
  private String name;

  public int getId() {
    return id;
  }

  public void setId(int id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  @NonNull
  @Override
  public String toString() {
    return super.toString();
  }
}

File: app/src/main/java/com/example/ruru/sophiesblog/java/collection/sort/Student.kt
package com.example.ruru.sophiesblog.java.collection.sort


data class Student(val id: Int, val name: String, val type: Int): Comparable<Student> {

  //降序排序
  override fun compareTo(other: Student): Int {
    return other.type - this.type
  }
}